astropy/wcs/utils.py

```diff
@@ -1056,7 +1056,7 @@ def fit_wcs_from_points(xy, world_coords, proj_point='center',
         raise ValueError("sip_degree must be None, or integer.")
 
     # set pixel_shape to span of input points
-    wcs.pixel_shape = (xp.max()+1-xp.min(), yp.max()+1-yp.min())
+    wcs.pixel_shape = np.ceil(np.max(xy, axis=1)).astype(np.int) + 1
```

Comment:
Suppose the `xy` locations are near the center. If you really want to limit the solution to the area in the image including the sources, then you need to take xy.min() into account. However, in this case the FITS WCS does not have a mechanism to set a domain where the solution is valid, something equivalent to the `bounding_box` in gwcs.

Presumably `xy` come from an image. Ideally `wcs.pixel_shape` should be set to the size of that image but the information is missing.

I wonder if this is one of those cases when `wcs.pixel_shape` should be left as `None`.

Reply:
> Suppose the `xy` locations are near the center. If you really want to limit the solution to the area in the image including the sources, then you need to take xy.min() into account.

That would be neede for the `bounding_box` but 
`wcs.pixel_shape` does not have the same meaning. 

> Presumably `xy` come from an image. Ideally `wcs.pixel_shape` should be set to the size of that image but the information is missing.
> 
> I wonder if this is one of those cases when `wcs.pixel_shape` should be left as `None`.

I think we should do our best to estimate this size. All we can do is to estimate minimal size by taking `max(xi)` (actual image may be larger but not have sources beyond `max(xi)`). I think that's fine. Current algorithm would under-estimate severily `wcs.pixel_shape` if sources are all bunched together even if all sources are near the xmax, ymax,... corner. Plus, CRPIX probably would be outside of the "image" defined by `wcs.pixel_shape` (which also should be taken at the center of the `wcs.pixel_shape` instead of being mean of positions, I think).